daemonset.yaml

```diff
@@ -0,0 +1,155 @@
+kind: ServiceAccount
+metadata:
+  name: storageos-daemonset-sa
+  namespace: default
+
+---
+
+# ClusterRole for init container.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: init-container
+rules:
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - create
+  - get
+  - list
+  - delete
+
+---
+
+# Bind DaemonSet ServiceAccount with init-container ClusterRole.
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: init-container
+subjects:
+- kind: ServiceAccount
+  name: storageos-daemonset-sa
+  namespace: default
+roleRef:
+  kind: ClusterRole
+  name: init-container
+  apiGroup: rbac.authorization.k8s.io
+
+---
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: storageos-daemonset
+  labels:
+    app: storageos-daemonset
+spec:
+  selector:
+    matchLabels:
+      name: storageos-daemonset
+  template:
+    metadata:
+      labels:
+        name: storageos-daemonset
+    spec:
+      serviceAccountName: storageos-daemonset-sa
+      initContainers:
+      - name: storageos-init
+        image: storageos/init:test
+        env:
+          - name: DAEMONSET_NAME
+            value: storageos-daemonset
+          - name: DAEMONSET_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+        volumeMounts:
+          - name: kernel-modules
+            mountPath: /lib/modules
+            readOnly: true
+          - name: sys
+            mountPath: /sys
+            mountPropagation: Bidirectional
+          - name: state
+            mountPath: /var/lib/storageos
+            mountPropagation: Bidirectional
+        securityContext:
+          privileged: true
+          capabilities:
+            add:
+            - SYS_ADMIN
+      containers:
+      - name: storageos
+        image: storageos/node:1.3.0
+        imagePullPolicy: IfNotPresent
+        args:
+          - server
+        ports:
+        - containerPort: 5705
+          name: api
+        livenessProbe:
+          initialDelaySeconds: 65
+          timeoutSeconds: 10
+          failureThreshold: 5
+          httpGet:
+            path: /v1/health
+            port: api
+        readinessProbe:
+          initialDelaySeconds: 65
+          timeoutSeconds: 10
+          failureThreshold: 5
+          httpGet:
+            path: /v1/health
+            port: api
+        env:
+        - name: HOSTNAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: JOIN
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: ADVERTISE_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: DISABLE_TELEMETRY
```

Comment:
Is this just for testing?

I don't think we'd want to start containers with telemetry disabled automatically.

Reply:
No, this manifest is only for testing. It's used in e2e test.

Reply:
Nice! In that case, thanks for disabling it during testing :)